analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ToStringShouldNotReturnNull.cs

```diff
@@ -0,0 +1,106 @@
+﻿public static class Condition
+{
+    public static bool When()
+    {
+        return true;
+    }
+}
+
+namespace Compliant
+{
+    class OtherMethodReturnsNullString
+    {
+        string Returns()
+        {
+            return null;
+        }
+    }
+
+    class ReturnsSomeString
+    {
+        public override string ToString()
+        {
+            if (Condition.When())
+            {
+                return "Hello, world!";
+            }
+            return "Hello, world!";
+        }
+    }
+
+    class ReturnsEmptyString
+    {
+        public override string ToString()
+        {
+            if (Condition.When())
+            {
+                return "";
+            }
+            return "";
+        }
+    }
+
+    class ReturnsStringEmpty
+    {
+        public override string ToString()
+        {
+            if (Condition.When())
+            {
+                return string.Empty;
+            }
+            return string.Empty;
+        }
+    }
+
+    struct ReturnsStringEmptyStruct
+    {
+        public override string ToString()
+        {
+            if (Condition.When()) { return string.Empty; }
+            return string.Empty;
+        }
+    }
+}
+
+class Noncompliant
+{
+    public class ReturnsNull
+    {
+        public override string ToString()
+        {
+            if (Condition.When())
+            {
+                return null; // Noncompliant
+            //  ^^^^^^^^^^^^
+            }
+            return null; // Noncompliant {{Return an empty string instead.}}
+        }
+    }
+
+    public class ReturnsNullViaExpressionBody
+    {
+        public override string ToString() => null; // Noncompliant
+    }
+
+    public class ReturnsNullViaTernaryExpressionBody
+    {
+        public override string ToString() => Condition.When() ? null : string.Empty; // Noncompliant
```

Comment:
Let's add UT with nested too. Should not be hard to support with some recursion in place

Reply:
Do you plan to address this?

Reply:
I had no clear example in mind, do you? (If so, I'm willing to add it)

Reply:
Just mechanical nesting, cosindering `Condition.When()` represents nondeterministic condition
```
public override string ToString() => 
    Condition.When() 
        ? (Condition.When() ? null : "something") // Noncompliant
         : (Condition.When() ? "something" : null) // Noncompliant;
```
Same as
```
public override string ToString() => 
    Condition1
        ? (Condition2 ? null : "something") // Noncompliant
         : (Condition3 ? "something" : null) // Noncompliant;
```

Reply:
It reports on the return (or arrow). I think that's fine, as changing the reporting node(s) is a lot of extra work, with limited value.

Reply:
Its fine to keep the node, no need to change that.